### PR TITLE
Pull Request: Basic Functionality of Driver service

### PR DESCRIPTION
**Pull request by katsiaryne**

### CODE REVIEW

File: src/main/kotlin/com/software/modsen/driverservice/dto/request/CarRequest.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+fun CarRequest.toCar(): Car {
+    return Car(
+        carId = null,
+        color = this.color,
+        model = this.model,
+        licensePlate = this.licensePlate,
+        year = this.year
+    )
+}
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/dto/request/CarRequest.kt]: Minor: even if its extension function for mapping - separate package is better

Comment (psidarovich) [reply src/main/kotlin/com/software/modsen/driverservice/dto/request/CarRequest.kt]: Also you can check scope functions:
https://kotlinlang.org/docs/scope-functions.html

File: src/main/kotlin/com/software/modsen/driverservice/dto/request/DriverRequest.kt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package com.software.modsen.driverservice.dto.request
+
+import com.software.modsen.driverservice.model.Driver
+import lombok.AllArgsConstructor
+import lombok.Getter
+import lombok.Setter
+
+@Getter
+@Setter
+@AllArgsConstructor
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/dto/request/DriverRequest.kt]: You don't need lombok when working with Kotlin data classes. 
Data class automatically generates toString, equals and hashCode functions with getters and setters for fields

File: src/main/kotlin/com/software/modsen/driverservice/exception/CarNotFoundException.kt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+package com.software.modsen.driverservice.exception
+
+class CarNotFoundException(s: String) : RuntimeException(s) {
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/exception/CarNotFoundException.kt]: Minor: if empty, remove redundant braces

File: src/main/kotlin/com/software/modsen/driverservice/model/Car.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+fun Car.toCarResponse(): CarResponse {
+    return CarResponse(
+        carId = this.carId,
+        color = this.color,
+        model = this.model,
+        licensePlate = this.licensePlate,
+        year = this.year
+    )
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/model/Car.kt]: Same here. Mapping functions should not be in such places

File: src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+@Service
+@RequiredArgsConstructor
+class DriverService(
+    private val driverRepository: DriverRepository,
+    private val carRepository: CarRepository
+) {
+    fun getDriverById(id: Long): DriverWithCarResponse {
+        return getByIdOrElseThrow(id).toDriverWithCarResponse()
+    }
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt]: Check expression functions, you can make your code even more readable

File: src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    fun createDriver(driverRequest: DriverRequest): DriverResponse {
+        preCreateValidateDriver(driverRequest)
+        val driver: Driver = driverRequest.toDriver()
+        driver.car = getCarByIdOrElseThrow(driverRequest.carId)
+        return driverRepository.save(driver).toDriverResponse()
+    }
+
+    fun updateDriver(id: Long, driverRequest: DriverRequest): DriverResponse {
+        val driver_opt: Driver = getByIdOrElseThrow(id)
```
Comment (psidarovich) [inline src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt]: Do not use snake case in kotlin or java apps

**Review by psidarovich — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/software/modsen/driverservice/dto/request/DriverRequest.kt (hunk 2 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/dto/request/DriverRequest.kt]: Removed unnecessary annotations

File: src/main/kotlin/com/software/modsen/driverservice/exception/CarNotFoundException.kt (hunk 3 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/exception/CarNotFoundException.kt]: Removed  redundant braces

File: src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt (hunk 6 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt]: Fixed

File: src/main/kotlin/com/software/modsen/driverservice/model/Car.kt (hunk 4 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/model/Car.kt]: Fixed

File: src/main/kotlin/com/software/modsen/driverservice/dto/request/CarRequest.kt (hunk 1 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/dto/request/CarRequest.kt]: Fixed

File: src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt (hunk 5 above)
Comment (katsiaryne) [reply src/main/kotlin/com/software/modsen/driverservice/service/DriverService.kt]: Changed to use extensions

**Review by psidarovich — APPROVED** [review]

**Pull request merged**